### ISSUE TITLE
Results' confidence

### ISSUE THREAD

**Issue by dberghi** [fuankarion/active-speakers-context#6]
Hi there!

Thank you for sharing your code and models, I really enjoyed your paper:)

I need to perform active speaker detection on audio-visual data for a project I'm working on. So I was looking for a sort of off-the-shelf way to do it and I found your work.
Therefore, I'm simply interested in applying your pre-trained models rather than training the models myself.

I know that in order to use the models on different data, not the AVA dataset, I will first have to perform the face thumbnails tracking and generate a csv file with similar tracking data as the ones provided with the AVA dataset. 
However, I've started by applying the models on a subset of the AVA dataset (the validation set) just to get familiar with your code and compare the results with the GT.   

This is briefly what I did:
- preprocess the val set with the given scripts to get the instance_crops and instance_wavs
- run STE_forward.py, loading the given STE.pth as model_weights (and adapting the path as required, including the config.py)
- run ASC_forward.py, loading the given ASC.pth as model_weights (and adapting the path as required, including the config.py)
- run ASC_predcition_postprocessing.py

It worked.
However, the results are not exactly as I was expecting, for example:

`
_7oWZq_s_Sk, 1440.4, 0.483333, 0.133333, 0.85, 0.7, SPEAKING_AUDIBLE, _7oWZq_s_Sk_1440_1500:1, 0.0277
`
Apart from the video name, time stamp, bbox coordinates, and entity id, that are the data taken from the gt files, the last number represents the prediction confidence, right?
So I would expect >0.5 = SPEAKING_AUDIBLE, <0.5 = NOT_SPEAKING, as, from my understanding, it comes from a softmax operation. 

Instead, I only get very low values, most of them 0.0*, sometimes 0.1 or 0.2. 
Additionally, comparing these confidence results with the GT they do not seem to follow similar patterns, so it is very difficult to identify a threshold.  

Is it correct? Did you get similar values too?

Thank you in advance for your answer and time.

**Comment by fuankarion** [comment]
Hi, are all the values that low? the AVA dataset is rather unbalanced about 60-70% of the labels are silence so you do expect a lot of small number in that column. Also don't get confused about the 'SPEAKING_AUDIBLE' tag, the evaluation tool requires it to be set to the same value. 
Anyway if you do find all the values to be that low, come back I'll try to reproduce it.
Btw, you could set 0.5 as a threshold for speech or silent, but you can also tune the value of the threshold for your task. Maybe you als want to check the output before the post-processing step. You will have two predictions, one for silence and one for speaking. You simply  compare them and go select the larger one or apply a soft-max over these values. The way you you turn the net predictions into a binary one really depends on your task so there is not one better way to do this.

**Comment by dberghi** [comment]
Hi, thank you for your reply.

The values in the columns before the post-processing step are always positive in the first column and always negative in the second. Same after the STE model: always positive in the column 'E' and negative in 'F'.
This is why the results of the softmax operation are always very low as if every entity is always silent.

Is there any chance that the problem is in the pretrained models?

Thanks!

**Comment by fuankarion** [comment]
Yeah, that looks like an issue with the weights. I'll check in a fresh install this weekend.

**Comment by dberghi** [comment]
Great, thank you for your time! Please keep me updated :)

**Comment by dberghi** [comment]
Hi there,
Have you got any news about the weights?
Thanks!

**Comment by fuankarion** [comment]
Hi, sorry I totally forgot about this, I'm getting back to it

**Comment by dberghi** [comment]
Hello! Sorry to bug you again.
Have you had a chance to recompute the weights?:) Thanks

**Comment by asker-github** [comment]
Hello, I didn't find a way to locate the sound. Do you known how to locate the sound in the frame.

**Comment by okankop** [comment]
Hi @dberghi, did you manage to get pretrained models working? I am trying the reproduce the results reported in the paper without success so far. I have trained the STE model from scratch for L=11, but it only achieved 66% mAP, but the same model achieves 79.5% mAP in the paper.

**Comment by dberghi** [comment]
Hi @okankop,
Eventually I retrained the model myself. Not sure whether the problem I had was due to something wrong I was doing or to the weights provided, as I asked when I opened this issue.
Anyway, I can confirm that everything worked fine. I used the default L=11, skip_frames=4 (for the ASC), and speakers=2 (for ASC, I don't remember what they used in the paper) 
For STE I got: train_mAP=0.812,  val_mAP=0.78

Are you using the training/val sets correctly? I have a total of 120 videos in the train set and 33 in the validation. Still not sure whether the test videos are released and available somewhere or still secret, I did not find them.

One other thing: check how the audio slices are organised in the parent folders. For example: /path/to/audio_slices/val/_7oWZq_s_SK_0900_0960:3/_7oWZq_s_SK_0900_0960:3.wav is wrong. It sould be: /path/to/audio_slices/val/_7oWZq_s_SK_0900_0960:3.wav
The preprocessing step produces empty directories and leaves the wav files outside. Check if your code is looking for the audio slices correctly. Otherwise your model will look for the audio files within empty directories, i.e it trains with visual-only data.

Hope this helped

**Comment by okankop** [comment]
@dberghi thanks for the detailed explanation. To get val mAP for STE, did you use the code [here](https://github.com/activitynet/ActivityNet/blob/master/Evaluation/get_ava_active_speaker_performance.py)?

**Comment by dberghi** [comment]
No, I checked in the log file generated during training. It's called STE_encoder_logs.csv. However I think you should be able to get it using the official evaluation tool released by the challenge's organisers too. I haven't tried

**Comment by okankop** [comment]
Thanks for the answer 👍

**Comment by Andrew-Brown1** [comment]
Hi @okankop ,

Did you find out why you only managed 66% mAP, I also only get training mAP to 66%

**Comment by SRA2** [comment]
Hi @dberghi how the ASC training progression looks like? I managed to train the STE and get 79.5 mAP. However, the ASC training starts with only ~21 mAP  and goes up to mAP 31 (number of speakers set to 2) with the features extracted using the STE trained model.

**Comment by SRA2** [comment]
> Hi @okankop ,
> 
> Did you find out why you only managed 66% mAP, I also only get training mAP to 66%
> 
> Thanks for any help.
> 
> EDIT: For reference my issue was with the audio data loader. If the path to the audio file does not exist, a zero tensor with an offset is still loaded. My audio data path was wrong. The first epoch results should look like this:
> 
> AV Loss: 0.4122 A Loss: 0.4527 V Loss: 0.5219 AUC: 0.8203 AP: 0.5196
> AV Loss: 0.3937 A Loss: 0.4344 V Loss: 0.5141 Acc: 101.8889 auROC: 0.8473 AP: 0.5725

Can you please report how the ASC training goes? like what is the first epoch's result?

**Comment by Leeviber** [comment]
> Hi @okankop,
> Eventually I retrained the model myself. Not sure whether the problem I had was due to something wrong I was doing or to the weights provided, as I asked when I opened this issue.
> Anyway, I can confirm that everything worked fine. I used the default L=11, skip_frames=4 (for the ASC), and speakers=2 (for ASC, I don't remember what they used in the paper)
> For STE I got: train_mAP=0.812, val_mAP=0.78
> 
> Are you using the training/val sets correctly? I have a total of 120 videos in the train set and 33 in the validation. Still not sure whether the test videos are released and available somewhere or still secret, I did not find them.
> 
> One other thing: check how the audio slices are organised in the parent folders. For example: /path/to/audio_slices/val/_7oWZq_s_SK_0900_0960:3/_7oWZq_s_SK_0900_0960:3.wav is wrong. It sould be: /path/to/audio_slices/val/_7oWZq_s_SK_0900_0960:3.wav
> The preprocessing step produces empty directories and leaves the wav files outside. Check if your code is looking for the audio slices correctly. Otherwise your model will look for the audio files within empty directories, i.e it trains with visual-only data.
> 
> Hope this helped

Hi, I met the same issue with you, due to lack of the GPU, I cannot retraining. Could you share your weights please?
